backend/github.js

```diff
@@ -96,3 +96,75 @@ exports.deleteHook = function (name, hook, accessToken, callback) {
     }
   });
 };
+
+/*
+ * hookValidator is used to check whether hook is registered to repository or not
+ * @param repositoryName: name of the repository
+ * @param token: github access token
+ */
+exports.hookValidator = function (repositoryName, token) {
+  return new Promise(function(resolve, reject) {
+    request({
+      url: `https://api.github.com/repos/${repositoryName}/hooks`,
+      headers: {
+        'User-Agent': 'Yaydoc',
+        'Authorization': 'token ' + crypter.decrypt(token)
+      }
+    }, function (error, response, body) {
+      var isRegistered = false;
+      if (response.statusCode !== 200) {
+        console.log(response.statusCode + ': ' + response.statusMessage);
+        reject()
+      }
+      var hooks = JSON.parse(body);
+      var hookurl = 'http://' + process.env.HOSTNAME + '/ci/webhook';
```

Comment:
Can't we use the current protocol of the browser by using just `//`? Like if the site is opened using http, use `http` else use `https`